### DISCUSSION TITLE
Integration with ORCID (email not available by default)

### DISCUSSION THREAD

**Discussion by holtgrewe** [fastapi-users/fastapi-users#1295] [Q&A] [answered]
I want to integrate with ORCID. This provider does not provide an email scope and retrieving the email of a user from their profile requires the users to explicite "allow access to email for trusted parties".

How deeply is the assumption of emails being available built into fastapi-users? As far as I can see, the `account_email` field is `nullable=False` in the model.

How would I go about adjusting the authorization flow to have users provide their email after login worked with ORCID but no email address was returned?

**Comment by frankie567** [comment, marked as answer]
Hi @holtgrewe 👋 

FastAPI Users assume an email address is available, as it's the unique identifier used for the user.

The easiest way IMO is to make your ORCID client return fake adresses when it's not available, like `user-{user_id}@orcid.org`. This way, you won't have to deeply change the behavior of the library. Users will then be able to change it if needed.
